biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/FastaStreamer.java

```diff
@@ -0,0 +1,179 @@
+package org.biojava.nbio.core.sequence.io;
+
+import org.biojava.nbio.core.sequence.ProteinSequence;
+import org.biojava.nbio.core.sequence.compound.AminoAcidCompound;
+import org.biojava.nbio.core.sequence.compound.AminoAcidCompoundSet;
+import org.biojava.nbio.core.sequence.io.template.SequenceCreatorInterface;
+import org.biojava.nbio.core.sequence.io.template.SequenceHeaderParserInterface;
+import org.biojava.nbio.core.util.InputStreamProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Read from a FASTA file (or gzipped FASTA file) and create a Java stream of {@link ProteinSequence} objects
+ * for use in a functional programming paradigm.
+ *
+ * @author Gary Murphy
+ * @since 7.0.3
```

Comment:
I think we should bump up to 7.1.0 since this introduces a (very nice) new feature

Reply:
I changed to 7.1.0-SNAPSHOT